tests/test_config_entries.py

```diff
@@ -4458,17 +4502,105 @@ async def test_avoid_starting_config_flow_on_single_config_entry(
         options={"vendor": "options"},
     )
     entry.add_to_hass(hass)
+    ignored_entry = MockConfigEntry(
+        domain="comp",
+        unique_id="2345",
+        title="Test",
+        data={"vendor": "data"},
+        options={"vendor": "options"},
+        source=config_entries.SOURCE_IGNORE,
+    )
+    ignored_entry.add_to_hass(hass)
 
     mock_platform(hass, "comp.config_flow", None)
 
     with patch(
         "homeassistant.loader.async_get_integration",
         return_value=integration,
-    ), pytest.raises(
-        HomeAssistantError,
-        match=r"Cannot start a config flow, the integration supports only a single config entry but already has one",
-    ):
-        await hass.config_entries.flow.async_init("comp", context={"source": "user"})
+    ), expectation:
+        result = await hass.config_entries.flow.async_init(
+            "comp", context={"source": source}, data=user_input
+        )
+
+    for key in expected_result:
+        assert result[key] == expected_result[key]
+
+
+@pytest.mark.parametrize(
+    ("source", "user_input", "expectation", "expected_result"),
+    [
+        (
+            config_entries.SOURCE_IGNORE,
+            {"unique_id": "blah", "title": "blah"},
+            does_not_raise(),
+            {"type": data_entry_flow.FlowResultType.CREATE_ENTRY},
+        ),
+        (
+            config_entries.SOURCE_REAUTH,
+            None,
+            does_not_raise(),
+            {"type": data_entry_flow.FlowResultType.FORM, "step_id": "reauth_confirm"},
```

Comment:
This case is a bit weird since the only existing config entry is an ignored one, but I guess it's technically ok.

Reply:
Yes, it's a bit weird maybe. I'm not opposed to removing it.

Reply:
Good to keep for now, I think.